src/nerf/websocket.gleam

```diff
@@ -15,14 +16,36 @@ pub type Frame {
   Binary(BitString)
 }
 
+pub type ConnectionOptsTransportOpts {
+  Verify(Atom)
+}
+
+pub type ConnectionOpts {
+  Transport(Atom)
+  TransportOpts(List(ConnectionOptsTransportOpts))
+}
+
 pub fn connect(
   hostname: String,
   path: String,
   on port: Int,
   with headers: List(Header),
 ) -> Result(Connection, ConnectError) {
+  let opts = []
+  connect_with_options(hostname, path, port, headers, opts)
+}
+
+pub fn connect_with_options(
+  hostname: String,
+  path: String,
+  on port: Int,
+  with headers: List(Header),
+  opts: List(ConnectionOpts),
```

Comment:
Oh this is interesting- the compiler should probably reject this as it's no longer possible to use the labels `on` and `with` as the following argument has no label. I'll open an issue for that in the compiler repo.